pkg/config/validation/validation.go

```diff
@@ -2067,8 +2068,13 @@ var ValidateVirtualService = registerValidateFunc("ValidateVirtualService",
 			}
 		}
 
+		isDelegate := false
 		if len(virtualService.Hosts) == 0 {
-			errs = appendErrors(errs, fmt.Errorf("virtual service must have at least one host"))
+			if features.EnableVirtualServiceDelegate.Get() {
+				isDelegate = true
```

Comment:
Having validation change at runtime is generally not a good idea. for example, I have this enabled, apply VS, then I disable it, and now I have a bunch of invalid config in the cluster

Reply:
I think this is not a big matter. Users are aware of their configs, they should know when to delete invalid ones.

Reply:
I think we can do intercept invalid ones in mergeVirtualServices

Reply:
Done